File: VB/CustomizeReportDataExample.Module/DatabaseUpdate/Updater.vb
Imports System
Imports System.Linq
Imports DevExpress.ExpressApp
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp.Xpo
Imports DevExpress.Persistent.BaseImpl
Imports CustomizeReportDataExample.Module.BusinessObjects

Namespace CustomizeReportDataExample.Module.DatabaseUpdate
    ' For more typical usage scenarios, be sure to check out http://documentation.devexpress.com/#Xaf/clsDevExpressExpressAppUpdatingModuleUpdatertopic
    Public Class Updater
        Inherits ModuleUpdater

        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
            MyBase.New(objectSpace, currentDBVersion)
        End Sub
        Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
            MyBase.UpdateDatabaseAfterUpdateSchema()
            If ObjectSpace.GetObjectsCount(GetType(Contact), Nothing) = 0 Then
                Dim mary As Contact = ObjectSpace.CreateObject(Of Contact)()
                mary.Name = "Mary"
                mary.Email = "mary@example.com"
                Dim john As Contact = ObjectSpace.CreateObject(Of Contact)()
                john.Name = "John"
                john.Email = "john@example.com"
            End If
            For Each reportData As MyReportDataV2 In ObjectSpace.GetObjects(Of MyReportDataV2)()
                If (reportData.IsPredefined) AndAlso String.IsNullOrEmpty(reportData.Category) Then
                    reportData.Category = "Predefined"
                End If
            Next reportData
        End Sub
        Public Overrides Sub UpdateDatabaseBeforeUpdateSchema()
            MyBase.UpdateDatabaseBeforeUpdateSchema()
            'if(CurrentDBVersion < new Version("1.1.0.0") && CurrentDBVersion > new Version("0.0.0.0")) {
            '    RenameColumn("DomainObject1Table", "OldColumnName", "NewColumnName");
            '}
        End Sub
    End Class
End Namespace

File: CS/CustomizeReportDataExample.Module/DatabaseUpdate/Updater.cs
using System;
using System.Linq;
using DevExpress.ExpressApp;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.ExpressApp.Xpo;
using DevExpress.Persistent.BaseImpl;
using CustomizeReportDataExample.Module.BusinessObjects;

namespace CustomizeReportDataExample.Module.DatabaseUpdate {
    // For more typical usage scenarios, be sure to check out http://documentation.devexpress.com/#Xaf/clsDevExpressExpressAppUpdatingModuleUpdatertopic
    public class Updater : ModuleUpdater {
        public Updater(IObjectSpace objectSpace, Version currentDBVersion) :
            base(objectSpace, currentDBVersion) {
        }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            if (ObjectSpace.GetObjectsCount(typeof(Contact), null) == 0) {
                Contact mary = ObjectSpace.CreateObject<Contact>();
                mary.Name = "Mary";
                mary.Email = "mary@example.com";
                Contact john = ObjectSpace.CreateObject<Contact>();
                john.Name = "John";
                john.Email = "john@example.com";
            }
            foreach (MyReportDataV2 reportData in ObjectSpace.GetObjects<MyReportDataV2>()) {
                if ((reportData.IsPredefined) && string.IsNullOrEmpty(reportData.Category)) {
                    reportData.Category = "Predefined";
                }
            }
        }
        public override void UpdateDatabaseBeforeUpdateSchema() {
            base.UpdateDatabaseBeforeUpdateSchema();
            //if(CurrentDBVersion < new Version("1.1.0.0") && CurrentDBVersion > new Version("0.0.0.0")) {
            //    RenameColumn("DomainObject1Table", "OldColumnName", "NewColumnName");
            //}
        }
    }
}
